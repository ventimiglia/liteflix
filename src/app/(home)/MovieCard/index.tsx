import Image from "next/image";
import { CATEGORY } from "../MovieList";

const MovieCard = ({
  movies,
  category,
}: {
  movies: Movie[];
  category: CATEGORY;
}) => (
  <>
    {movies?.length > 0 ? (
      movies.map((movie) => (
        <article
          key={movie.id}
          className="flex flex-col justify-between gap-4 w-80 xl:w-56 h-44 xl:h-36 relative bg-gradient-to-t from-gray-800 to-transparent rounded-sm p-3 xl:min-h-[9rem]"
        >
          <Image
            src={`${
              category === CATEGORY.POPULAR
                ? `https://image.tmdb.org/t/p/original${movie.backdrop_path}`
                : movie.backdrop_path
            }`}
            alt={movie?.title || "Movie title"}
            fill
            priority
            className="absolute -z-10 object-cover w-full h-full rounded-sm"
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 30vw, 360px"
          />
          <div className="w-10 h-10 border border-white rounded-full bg-transparent text-center self-center relative top-1/2 -translate-y-1/2">
            <Image
              src="/play.svg"
              alt="Reproducir"
              width={14}
              height={14}
              className="relative top-1/2 left-1/2 transform -translate-x-[35%] -translate-y-1/2"
            />
          </div>
          <h3 className="text-center w-full overflow-ellipsis whitespace-nowrap overflow-hidden">
            {movie?.title}
          </h3>
        </article>
      ))
    ) : (
      <p>No hay peliculas</p>
    )}
  </>
);

export default MovieCard;
