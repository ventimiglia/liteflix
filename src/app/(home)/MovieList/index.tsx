"use client";

import CustomDropdown from "@/components/Dropdown";
import { useState } from "react";
import MovieCard from "../MovieCard";
import { AnimatePresence } from "framer-motion";

type Props = {
  popularMovies: Movie[];
  myMovies: Movie[];
  className?: string;
};

export const enum CATEGORY {
  POPULAR = "POPULARES",
  MY_MOVIES = "MIS PELICULAS",
}

const MovieList = ({ popularMovies, myMovies, className }: Props) => {
  const [selectedCategory, setSelectedCategory] = useState(CATEGORY.POPULAR);
  const movies =
    selectedCategory === CATEGORY.POPULAR ? popularMovies : myMovies;
  return (
    <aside
      className={`lg:animate-fade-left lg:animate-once lg:animate-delay-1000 lg:animate-ease-linear lg:animate-normal lg:animate-fill-both flex-col z-0 items-center justify-start h-full self-center w-56 min-w-max py-6 gap-4 overflow-y-auto overflow-x-hidden lg:max-h-[46rem] ${className}`}
    >
      <CustomDropdown
        selectedCategory={selectedCategory}
        setSelectedCategory={setSelectedCategory}
      />
      <AnimatePresence>
        {movies.length === 0 ? (
          <p className="text-white text-center">No hay peliculas</p>
        ) : (
          <>
            {movies.map((movie) => (
              <MovieCard key={movie.id} movie={movie} selectedCategory={selectedCategory} />
            ))}
          </>
        )}
      </AnimatePresence>
    </aside>
  );
};

export default MovieList;
