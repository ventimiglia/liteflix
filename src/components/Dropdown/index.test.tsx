import React from "react";
// import '@testing-library/jest-dom'

import { render, fireEvent } from "@testing-library/react";
import CustomDropdown from "./";
import { CATEGORY } from "@/app/(home)/MovieList";

const defaultProps = {
  selectedCategory: CATEGORY.POPULAR,
  setSelectedCategory: jest.fn(),
};
describe("CustomDropdown", () => {
  describe("Rendering", () => {
    it("renders the dropdown button with selected category", () => {
      const { getByText } = render(<CustomDropdown {...defaultProps} />);
      const dropdownButton = getByText(CATEGORY.POPULAR);
      expect(dropdownButton).toBeInTheDocument();
    });
  });

  describe("Actions", () => {
    it("opens the dropdown menu when the button is clicked", () => {
      const { getByText } = render(<CustomDropdown {...defaultProps} />);
      const dropdownButton = getByText(CATEGORY.POPULAR);

      fireEvent.click(dropdownButton);

      const dropdownMenu = getByText(CATEGORY.MY_MOVIES);
      expect(dropdownMenu).toBeInTheDocument();
    });

    it("selects a category when an option is clicked", () => {
      const setSelectedCategory = jest.fn();
      const { getByText } = render(
        <CustomDropdown
          {...defaultProps}
          setSelectedCategory={setSelectedCategory}
        />
      );

      const dropdownButton = getByText(CATEGORY.POPULAR);
      fireEvent.click(dropdownButton);

      const myMoviesOption = getByText(CATEGORY.MY_MOVIES);
      fireEvent.click(myMoviesOption);

      expect(setSelectedCategory).toHaveBeenCalledWith(CATEGORY.MY_MOVIES);
    });
  });
});
