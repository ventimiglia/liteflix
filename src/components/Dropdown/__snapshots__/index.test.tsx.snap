// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDropdown Actions opens the dropdown menu when the button is clicked and check snapshot 1`] = `
<div>
  <div
    class="relative inline-block text-left z-10 w-full"
  >
    <button
      class="inline-flex justify-center items-center w-full rounded-md px-4 py-2 bg-transparent text-sm font-medium text-white transition-colors hover:text-white/50"
      type="button"
    >
      <span>
        VER: 
        <b>
          POPULARES
        </b>
      </span>
      <img
        alt="arrow"
        class="w-auto h-auto ml-2"
        data-nimg="1"
        decoding="async"
        height="13"
        loading="lazy"
        src="/arrow.svg"
        style="color: transparent;"
        width="13"
      />
    </button>
    <div
      class="origin-top-right absolute w-full right-0 mt-2 shadow-lg bg-secondary text-white before:content-[''] before:h-3 before:w-3 before:bg-secondary before:absolute before:-translate-y-1/2 before:right-[15%] before:rotate-45"
      style="opacity: 0;"
    >
      <div
        aria-labelledby="options-menu"
        aria-orientation="vertical"
        class="flex flex-col py-4 gap-4"
        role="menu"
      >
        <button
          class="flex items-center justify-between px-4 text-sm transition-colors hover:text-white/80 w-full text-left font-bold"
          role="menuitem"
        >
          POPULARES
          <img
            alt="arrow"
            data-nimg="1"
            decoding="async"
            height="13"
            loading="lazy"
            src="/check.svg"
            style="color: transparent;"
            width="13"
          />
        </button>
        <button
          class="flex items-center justify-between px-4 text-sm transition-colors hover:text-white/80 w-full text-left "
          role="menuitem"
        >
          MIS PELICULAS
        </button>
      </div>
    </div>
  </div>
</div>
`;
